internal/pkg/util/fs/overlay/overlay_linux.go

```diff
@@ -46,11 +45,6 @@ var incompatibleFs = map[int64]fs{
 		name:       "NFS",
 		overlayDir: upperDir,
 	},
-	// FUSE filesystem
-	fuse: {
```

Comment:
What problem exactly would it cause with kernel overlay?  I have observed that using fuse2fs as the upper dir causes an "invalid argument" error when attempting to start the unprivileged kernel overlay on el8, but with this PR it silently (other than a debug message, I believe added here) proceeded to use fuse-overlayfs.  Are you concerned about what would happen when using privileged overlay with a fuse2fs upper dir?  That seems to be an unlikely combination, but I just tested manually running fuse2fs.  First, trying as unprivileged user I got the error
```
FATAL:   container creation failed: while setting overlay session layout: only root user can use sandbox as overlay in setuid mode
```
Then I tried running it with sudo and got the error
```
FATAL:   while loading overlay images: failed to open overlay image /home/dwd/scratch/mnt: failed to retrieve path for /home/dwd/scratch/mnt: lstat /home/dwd/scratch/mnt: permission denied
```
Then I re-ran it after running fuse2fs as root and got the error
```
FATAL:   container creation failed: mount overlay->/var/apptainer/mnt/session/final error: while mounting overlay: can't mount overlay filesystem to /var/apptainer/mnt/session/final: invalid argument
```
Is that the scenario you're worried about, that the error message is not as helpful?   Are there other cases?

Reply:
> Are you concerned about what would happen when using suid-flow overlay with a fuse2fs upper dir? That seems to be an unlikely combination, but I just tested manually running fuse2fs

No, let's say that `/mnt/container_overlay` is any fuse filesystem mount point and someone attempt to do something like this as root :

```
apptainer run -o /mnt/container_overlay image.sif
```

This would return a cryptic error and only `dmesg` will tell what's wrong, kernel overlay doesn't allow FUSE as upper directory, actually apptainer detect that by aborting with an explicit message.
I understand why you remove it first, but this is not the correct way to achieve what you need to make fuse2fs working as an upper overlay

Reply:
Ok, I made that change.  Please re-review